go-slang/src/compiler/index.ts

```diff
@@ -155,7 +173,13 @@ export class GolangCompiler {
       );
       this.compile(assignAst);
     },
+    ChanType: (astNode: AST.ChanType) => {
+      throw new CompilationError("ChanType not handled");
+    },
     CallExpr: (astNode: AST.CallExpr) => {
+      if (astNode.Fun.Name === "make") {
+        return this.compile_make(astNode);
+      }
```

Comment:
In our `CallExpr` interface, `Fun` has a type of `Ident` which allows you to do `astNode.Fun.Name` and check if its `make`.

However, the actual type should be `Expr` according to Go docs: 
![image](https://github.com/xlzior/goats/assets/46375597/0f97e9c4-a6fd-4e97-b95e-f943626bc963)

This would cause problems if I change the type to `Expr` to implement `mutex.Lock()`/`Unlock()`, where `Fun` has a type `SelectorExpr`. TypeScript would complain that `Name` doesn't exist on `Expr` hahaha. 

Anyway, for mutex.Lock()/Unlock(), even if I change the type to `Expr`, I would have to compile `Fun` (SelectorExpr) first and then add a CALL instr later. this would still be problematic because I want to handle Lock() differently from a normal function call. Maybe i just implement Lock() and Unlock() in a non-OOP way and follow the same pattern as u?